compiler/qsc/src/interpret/stateful/stepping_tests.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::interpret::stateful::Interpreter;
+use qsc_eval::{output::CursorReceiver, StepAction, StepResult};
+use qsc_fir::fir::StmtId;
+use qsc_frontend::compile::SourceMap;
+use qsc_passes::PackageType;
+use std::io::Cursor;
+
+#[cfg(test)]
+mod given_interpreter_with_sources {
+    use super::*;
+
+    static STEPPING_SOURCE: &str = r#"
+        namespace Test {
+            @EntryPoint()
+            operation A() : Int {
+                let d = B();
+                let e = d / 1;
+                e
+            }
+            operation B() : Int {
+                let g = 10;
+                let h = 20;
+                let l = C(g, h);
+                42
+            }
+            operation C(m: Int, n: Int) : Int {
+                let o = 42 - (m + n);
+                let p = (m + n) + o;
+                p
+            }
+        }"#;
```

Comment:
Do we have any test-cases that include non-ascii chars - being that we had an issue with those that required the mapping stuff be added. Might be a good safety net.

Reply:
The offsets are never used when running. We use distinct int ids for breakpoints. The utf8=>utf16 is only done in the JS layer for stack frames and breakpoint ui location mapping. The spans are never sent back into the compiler.